### PR TITLE
feat: add builds for node 18

### PR DESCRIPTION
**Pull request by Rauno56**
# Description

Adds builds for node@18.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Internal change (a change which is not visible to the package consumers)
- [ ] Documentation change or requires a documentation update

# How Has This Been Tested?

- [ ] Tested manually
- [ ] Added automated tests

# Checklist:

- [ ] Unit tests have been added/updated
- [ ] Documentation has been updated
- [ ] Change file has been generated (`npm run change:new`)
- [ ] Delete this branch (after the PR is merged)

### CODE REVIEW

File: binding.gyp [hunk 1]
```diff
@@ -18,7 +18,7 @@
         "cflags": [
           "-std=c++11",
           "-Wall",
-          "-Werror"
```
Comment (seemk) [inline binding.gyp:21]: Possible to keep `-Werror` alongside `-Wno-deprecated`?

Comment (seemk) [reply binding.gyp:21]: `-Wno-deprecated` will disable deprecation warnings, `-Werror` will still keep other warnings as errors

Comment (Rauno56) [reply binding.gyp:21]: new versions of `nan` and `node-abi` are now released. Changed the build arguments back.

**Comment by Rauno56** [comment]
@seemk, PTAL.

**Review by mhennoch — APPROVED** [review]

**Review by seemk — APPROVED** [review]

**Pull request merged**